CreeDictionary/DatabaseManager/cree_inflection_generator.py

```diff
@@ -46,41 +23,39 @@ def expand_inflections(
     """
     logger = DatabaseManagerLogger(__name__, verbose)
 
-    # optimized for efficiency by calling hfstol once and for all
-    flattened_layouts = import_flattened_prefilled_layouts(
-        Path(dirname(__file__)) / ".." / "res" / "prefilled_layouts"
-    )
-    fat_generated_analyses = []
+    paradigm_filler = ParadigmFiller.default_filler()
 
     analyses = list(analyses)
-
-    to_generated = dict()  # type: Dict[str, List[str]]
+    to_generated: Dict[str, List[str]] = {}
+    # We'll generate all of the forms for analyses enqueued here in one fell swoop.
+    analysis_queue = []
 
     for analysis in analyses:
         lemma_category = fst_analysis_parser.extract_lemma_text_and_word_class(analysis)
         assert lemma_category is not None
         lemma, category = lemma_category
-        if category.is_verb() or category.is_noun():
-            generated_analyses = [
-                generated_analysis.replace("{{ lemma }}", lemma)
-                for generated_analysis in flattened_layouts[category]
-            ]
+
+        if category.has_inflections():
+            generated_analyses = list(paradigm_filler.expand_analyses(lemma, category))
         else:
             generated_analyses = [analysis]
+
         to_generated[analysis] = generated_analyses
-        fat_generated_analyses.extend(generated_analyses)
```

Comment:
I liked the name "fat_generated_analyses", but I imagine it more as a queue of analyses ready to be inflected by the generator FST.